src/Altinn.App.Core/Models/Layout/Components/OptionsComponent.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections.Immutable;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+using Altinn.App.Core.Models.Expressions;
+
+namespace Altinn.App.Core.Models.Layout.Components;
+
+/// <summary>
+/// Custom component for handeling the special fields that represents an option.
+/// </summary>
+public class OptionsComponent : BaseComponent
+{
+    /// <summary>
+    /// The ID that references <see cref="Altinn.App.Core.Features.IAppOptionsProvider.Id" /> and <see cref="Altinn.App.Core.Features.IInstanceAppOptionsProvider.Id" />
+    /// </summary>
+    public string? OptionId { get; }
+
+    /// <summary>
+    /// Alternaltive to <see cref="OptionId" /> where the options are listed inline instead of referencing an external generator
+    /// </summary>
+    public List<AppOption>? Options { get; }
+
+    /// <summary>
+    /// Is the component referencing a secure code list (uses security context of the instance)
+    /// </summary>
+    public bool Secure { get; }
+
+    /// <summary>
+    /// Constructor
+    /// </summary>
+    public OptionsComponent(string id, string type, IReadOnlyDictionary<string, string>? dataModelBindings, Expression? hidden, Expression? required, Expression? readOnly, string? optionId, List<AppOption>? options, bool secure, IReadOnlyDictionary<string, string>? extra) :
```

Comment:
Does it make sense to group some of the parameters into a parameter object to reduce the number and make it a bit more readable? hidden, required, readonly could be candidates for grouping.

Reply:
Maybe? but they will be separate variables in the parser, and separate properties on the Component class. I don’t see the huge benefit of grouping them before calling the constructor.

Reply:
I made an attempt, but I'm not sure it improved readability to a huge extent. https://github.com/Altinn/app-lib-dotnet/commit/225027b2f200706c10473592a70bf69d66a8ff7d

Reply:
I think the reduced number of parameters in the constructor made it more readable, although I was a bit confused at first since you used the component base class itself. I was more thinking about a simple parameter object if any parameters naturally belonged together and could be named thereafter eg. DisplayOptions (hidden, required, readonly) or something. Personally, I strive to keep the number of parameters to 3 or less, max 5, but here we have 10 which I hoped we could reduce a bit. But won't block any approval:)

Reply:
Given your vague reply, I think I'd like to keep the current constructor.